community/rfcs/techniques/proposals/20200205-weight-clustering-proposal.md

```diff
@@ -0,0 +1,82 @@
+# Keras Weight Clustering API
+
+| Status        | Accepted                                      |
+:---------------|:----------------------------------------------|
+| **Author(s)** | Mohamed Nour Abouelseoud (mohamednour.abouelseoud@arm.com), Aron Virginas-Tar (aron.virginas-tar@arm.com), Anton Kachatkou (anton.kachatkou@arm.com)|
+| **Sponsor**   | Alan Chiao (alanchiao@google.com)             |
+| **Updated**   | 2020-03-06                                    |
+
+## Objective
+
+This RFC document addresses the technique &ndash; and consequent API change &ndash; of weight clustering. This is an approach for compressing neural networks by restricting the number of unique weights so that a lower number of bits is required to represent each weight.
+
+## Motivation
+
+Clustering, or weight sharing, allows for certain types of hardware to benefit from advanced weight compression techniques and the associated reduction in model memory footprint and bandwidth.
```

Comment:
To summarize the benefits, start with what is generalizable across hardware and backends. Starting with "certain types of hardware" gives the impression that non of the benefits are generalizable. 

We can always get storage / over-the-wire size reductions via zip.  Sample:

This technique brings improvements via model compression. Further framework and hardware support can result in reductions in memory footprint and memory bandwidth.

If you think "model compression" is ambiguous, we can replace that with "by reducing model storage and over-the-wire size" or something similar. 

"bandwidth" -> "memory bandwidth" since someone might think of it as "network bandwidth". Also we would be improving memory bandwidth and not reducing it right? 
It doesn't need to be in this proposal, but how does weight clustering improve the number of bytes/second for the transfer (my understanding of what bandwidth means)? I can see weight clustering reducing the total number of bytes to transfer, but bytes/second is not obvious.